Fix accept4 declaration in uClibc when not available in Linux Kernel.
This patch is useful to make new Qt versions (>= 4.8) compile with old Linux kernel (< 2.6.38).

Signed-off-by: Jérémie Scheer <jeremie.scheer@armadeus.com>

Index: buildroot/package/uclibc/0.9.33.2/uclibc-0055-accept4-declaration-fix.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/uclibc/0.9.33.2/uclibc-0055-accept4-declaration-fix.patch	2014-02-17 16:13:26.000000000 +0100
@@ -0,0 +1,33 @@
+From db6b038ff049a5b6e11b03d2cdb66248002eafff Mon Sep 17 00:00:00 2001
+From: Guillaume Emont <guijemont@igalia.com>
+Date: Thu, 15 Sep 2011 20:19:19 +0200
+Subject: [PATCH 1/2] Only expose accept4 in headers if it is available
+
+---
+ include/sys/socket.h |    3 ++-
+ 1 files changed, 2 insertions(+), 1 deletions(-)
+
+diff --git a/include/sys/socket.h b/include/sys/socket.h
+index 6037f92..53b56ea 100644
+--- a/include/sys/socket.h
++++ b/include/sys/socket.h
+@@ -22,6 +22,7 @@
+ #define	_SYS_SOCKET_H	1
+
+ #include <features.h>
++#include <bits/kernel-features.h>
+
+ __BEGIN_DECLS
+
+@@ -227,7 +228,7 @@ extern int accept (int __fd, __SOCKADDR_ARG __addr,
+ 		   socklen_t *__restrict __addr_len);
+ libc_hidden_proto(accept)
+
+-#if defined __UCLIBC_LINUX_SPECIFIC__ && defined __USE_GNU
++#if defined __UCLIBC_LINUX_SPECIFIC__ && defined __USE_GNU && defined __ASSUME_ACCEPT4
+ /* Similar to 'accept' but takes an additional parameter to specify flags.
+
+    This function is a cancellation point and therefore not marked with
+--
+1.7.4.1
+
Index: buildroot/package/uclibc/0.9.33.2/uclibc-0056-assume-accept4-in-recent-arm-kernels.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/uclibc/0.9.33.2/uclibc-0056-assume-accept4-in-recent-arm-kernels.patch	2014-02-17 16:13:41.000000000 +0100
@@ -0,0 +1,28 @@
+From c526492ee2284b6ae61040b2eac5b03120fb3185 Mon Sep 17 00:00:00 2001
+From: Guillaume Emont <guijemont@igalia.com>
+Date: Thu, 15 Sep 2011 20:19:57 +0200
+Subject: [PATCH 2/2] define __ASSUME_ACCEPT4 for arm for kernels >= 2.6.36
+
+---
+ libc/sysdeps/linux/common/bits/kernel-features.h |    5 +++++
+ 1 files changed, 5 insertions(+), 0 deletions(-)
+
+diff --git a/libc/sysdeps/linux/common/bits/kernel-features.h b/libc/sysdeps/linux/common/bits/kernel-features.h
+index 6bf5544..be92f90 100644
+--- a/libc/sysdeps/linux/common/bits/kernel-features.h
++++ b/libc/sysdeps/linux/common/bits/kernel-features.h
+@@ -479,6 +479,11 @@
+ # define __ASSUME_ACCEPT4       1
+ #endif
+
++/* Correct support for the accept4 syscall for arm was added in 2.6.36.  */
++#if __LINUX_KERNEL_VERSION >= 0x020624 && defined __arm__
++# define __ASSUME_ACCEPT4       1
++#endif
++
+ /* Support for the FUTEX_CLOCK_REALTIME flag was added in 2.6.29.  */
+ #if __LINUX_KERNEL_VERSION >= 0x02061d
+ # define __ASSUME_FUTEX_CLOCK_REALTIME	1
+--
+1.7.4.1
+
